src/server/services/wms/qgswmsparameters.cpp

```diff
@@ -1656,36 +1690,48 @@ namespace QgsWms
     return style << styles;
   }
 
-  QList<QgsWmsParametersLayer> QgsWmsParameters::layersParameters() const
+  QMultiMap<QString, QString> QgsWmsParameters::getLayerFilters( const QStringList &layers ) const
   {
-    QList<QgsWmsParametersLayer> parameters;
-    QStringList layers = allLayersNickname();
-    QStringList styles = allStyles();
-    QStringList filter = filters();
-    QStringList selection = selections();
-    QList<int> opacities = opacitiesAsInt();
-
     // filter format: "LayerName:filterString;LayerName2:filterString2;..."
     // several filters can be defined for one layer
+    const QStringList rawFilters = filters();
     QMultiMap<QString, QString> layerFilters;
-    Q_FOREACH ( QString f, filter )
+    for ( int i = 0; i < rawFilters.size(); i++ )
```

Comment:
Maybe we could use something like:

``` c++
for ( const QString &rawFilter : rawFilters )
```

Reply:
No, I need `i` for being able to access `layers[i]`

Reply:
Indeed, I didn't see that...